### DISCUSSION TITLE
Put avo related things in one folder

### DISCUSSION THREAD

**Discussion by jony1993** [avo-hq/avo#649] [General]
At the moment the different Avo parts are in different folders, spread over the whole project. 
Maybe it would be awesome to have everything avo related somehow in one avo subfolder.

e.g.
avo:
- actions
- resources
- controllers
- policies
- views

That would help to keep the Admin Part seperated from the main application.

**Comment by adrianthedev** [comment]
I am all for having things organized.
Right now we have the following:

```
- app
  - avo
    - actions
    - fields
    - filters
    - resources
  - controllers
    - avo
      - ... resource controllers
  - views
    - avo
      - tools
      - ... misc partials
```

I followed what others are doing (devise puts it's views in the views dir)

TBH, the scheme you are proposing doesn't sound that bad. Are you aware of other gems that use that kind of file organization?

**Comment by jony1993** [comment]
Hi Adrian,

Sorry for the late reply.
I dont know other gems which are doing it like my suggestion. I'll search a bit...

I just thought, that it might make sense to have everything in one place. That would not mix our main core app with the admin stuff...

**Reply by adrianthedev** [reply]
Yes, that makes sense (not mixing things together). I'm a bit concerned about pulling things away from "their places". Especially views and controllers. Also, where would policies go?

**Reply by adrianthedev** [reply]
Just to be clearer, I'd like to do that, separate the files.

I'm wondering if we could support both scenarios (separate and merged). Will have to look into it.
